### DISCUSSION TITLE
response headers for account recovery link

### DISCUSSION THREAD

**Discussion by ujaved-doc-ai** [ory/kratos#2228] [Q&A] [answered]
I'm trying to implement [account recovery](https://www.ory.sh/kratos/docs/self-service/flows/account-recovery/). Basically when the user clicks on the recovery link, the kratos response header is as expected, with 303 redirection and the set-cookie header for `ory_kratos_session`:
![image](https://user-images.githubusercontent.com/94198560/153951475-b75570a9-f0fa-481c-ad2a-904fb0947062.png)

However in the same browser when a different user clicks on their own recovery link, the kratos response header doesnt seem to have a new cookie for `ory_kratos_session`. Instead the redirect request has the `Cookie` header set from the previous link's response:
![image](https://user-images.githubusercontent.com/94198560/153952001-d321b6c2-711d-4152-b4e3-5fd3467c9281.png)

it seems like the browser is re-using the cookie from the previous link and also kratos isn't sending the new cookie in the response header (even though `no-cache` is set)t? Is my interpretation correct? Is there a way we can maybe change the cache settings so that we get the new `ory_kratos_session` everytime?

**Comment by aeneasr** [comment, marked as answer]
https://github.com/ory/kratos/pull/2195 :)
